api/api.go

```diff
@@ -256,11 +267,95 @@ func CreateNamespace(ctx context.Context, logger *logger.Logger, name string, or
 		return nil, err
 	}
 
-	namespace, err := createNamespaceByID(ctx, logger, name, organizationID)
+	namespace, err := createNamespaceWithOwnerID(ctx, logger, name, organizationID)
 
 	if err != nil {
 		return nil, err
 	}
 
 	return namespace, err
 }
+
+func getNamespace(ctx context.Context, logger *logger.Logger, name string) (string, error) {
+	var response struct {
+		RegistryNamespace struct {
+			ID string
+		}
+	}
+
+	query := `
+				query($name: String!) {
+					registryNamespace(
+						name: $name
+					){
+						id
+					}
+			 }`
+	request := client.NewAuthorizedRequest(viper.GetString("token"), query)
+	request.Var("name", name)
+
+	graphQLclient := client.NewClient(viper.GetString("endpoint"), logger)
+
+	err := graphQLclient.Run(ctx, request, &response)
+
+	if err != nil || response.RegistryNamespace.ID == "" {
+		err = fmt.Errorf("Unable to find namespace %s", name)
+	}
```

Comment:
A couple of comments here:

- Let's split the case of when `err` is nil from the checking of `response.RegistryNamespace.ID`. We want to either return the error, or a wrapper around the error when it is not null.
- When can `response.RegistryNamespace.ID` be empty/null? That seems like a fundamental failure in the API schema.